s6100/scripts/s6100_platform.sh

```diff
@@ -16,38 +16,74 @@ init_devnum() {
     [ $found -eq 0 ] && echo "cannot find iSMT" && exit 1
 }
 
+# Perform an i2c device configuration : instantiate / delete.
+# Input is of the form:
+# "echo [driver] <i2c-address> >  <i2c-bus/operation>"
+# where operation = "new_device" or "delete_device"
+
+i2c_config() {
+    local count=0
+    local MAX_BUS_RETRY=20
+    local MAX_I2C_OP_RETRY=10
+
+    i2c_bus_op=`echo "$@" | cut -d'>' -f 2`
+    i2c_bus=$(dirname $i2c_bus_op)
+
+    # check if bus exists
+    while [ "$count" -lt "$MAX_BUS_RETRY" ]; do
+        [ -e $i2c_bus ] && break || sleep .1
```

Comment:
The sleep was added initially since it is perceived that the mux instantiation takes a non 0 time (atleast a second) to create the downstream i2c buses. I had not been able to hit a condition where sleep was necessary earlier.

Currently, in all my the testing thusfar (including on PTF testbed of Z9100 & S6100), I have not seen that the sleep/retry code is hit (based on checking the count).

If you perceive this to be redundant, will remove.